### PR TITLE
Better error messaging when type is set as a non-string/numeric

### PR DESCRIPTION
**Pull request by andrewvc**
There are various user errors as a result of 'type' being set to non-string values (this is a particular problem with the java plugin), see #63 and #138 . This patch will provide much better messaging to the user. We should migrate this to the -java version later.

### CODE REVIEW

File: lib/logstash/outputs/elasticsearch.rb [hunk 1]
```diff
@@ -58,7 +58,7 @@ class LogStash::Outputs::ElasticSearch < LogStash::Outputs::Base
   # similar events to the same 'type'. String expansion `%{foo}` works here.
   #
   # Deprecated in favor of `document_type` field.
-  config :index_type, :validate => :string, :deprecated => "Please use the 'document_type' setting instead. It has the same effect, but is more appropriately named."
+  config :index_type, :validate => :string, :obsolete => "Please use the 'document_type' setting instead. It has the same effect, but is more appropriately named."
```
Comment (untergeek) [inline lib/logstash/outputs/elasticsearch.rb]: HUGE :+1:  here!

File: lib/logstash/outputs/elasticsearch.rb [hunk 2]
```diff
@@ -343,6 +334,23 @@ def receive(event)
     buffer_receive([event.sprintf(@action), params, event])
   end # def receive
 
+  def get_event_type(event)
```
Comment (untergeek) [inline lib/logstash/outputs/elasticsearch.rb]: I like that this is a method now.  So much easier to do unit testing!

**Comment by untergeek** [comment]
So far it LGTM™

**Comment by untergeek** [comment]
LGTM™

"So long and thanks for all the tests..."

**Comment by purbon** [comment]
please jenkins, test this.

**Comment by andrewvc** [comment]
please jenkins, test this

**Comment by andrewvc** [comment]
please jenkins, test this

**Comment by talevy** [comment]
LGTM

**Pull request closed without merging**